### DISCUSSION TITLE
Sharing View Component Previews Across Multiple Projects Without Duplicating Files

### DISCUSSION THREAD

**Discussion by dlupu** [lookbook-hq/lookbook#607] [General]
Hello,

We currently have an internal gem that facilitates the sharing of a collection of several dozen view components across multiple projects.

The final appearance of these components is project-specific, with each project featuring its own unique design system.

We're exploring potential solutions to share previews and scenarios for these components throughout all the projects. Our aim is to view them in the project's lookbook (essentially rendering them using the project's CSS) without the need to duplicate any files.

Any ideas or suggestions on how to best approach this situation ? Is this something you would consider supporting ?

**Comment by allmarkedup** [comment]
Hey @dlupu - this should be possible, you could include the component previews in the shared gem, and then have a Lookbook instance per-project that points to the preview files in the Gem.

In the config for each project you could point to the gem's preview files directory using something like this:

```rb
gem_path = Gem.loaded_specs[NAME_OF_GEM].full_gem_path
config.lookbook.preview_paths = "#{gem_path}/path/to/previews"
```

Something like that should work if I have understood your question correctly?
